src/player.c

```diff
@@ -2167,11 +2173,89 @@ playback_next_bh(void *arg, int *retval)
   return COMMAND_END;
 }
 
+static int
+seek_calc_position_ms(struct player_seek_param *seek_param, struct db_queue_item **queue_item, int *position_ms)
+{
+  struct db_queue_item *seek_queue_item = NULL;
+  int seek_ms = 0;
+
+  // Initialize out parameters
+  *queue_item = NULL;
+  *position_ms = 0;
+
+  // Calculate seek position
+  if (seek_param->mode == PLAYER_SEEK_POSITION)
+    seek_ms = seek_param->ms;
+  else
+    seek_ms = pb_session.playing_now->pos_ms + seek_param->ms;
+
+  // Check if we need to switch to a previous track, this will be done if we are in the first 3 seconds
+  // of a track and we have a seek request for more than 3 seconds
+  if (seek_ms < 0)
+    {
+      if (pb_session.playing_now->pos_ms < 3000)
+	{
+	  // We are in the first 3 seconds of the track, switch to the previous track and recalculate the absolute seek position
+	  seek_queue_item = queue_item_prev(pb_session.playing_now->item_id);
+
+	  if (seek_queue_item)
+	    {
+	      seek_ms = seek_queue_item->song_length + seek_ms;
+	      // Make sure to not try to seek behind the previous track (this is also the case if song_length is zero)
+	      seek_ms = (seek_ms < 0) ? 0 : seek_ms;
+	    }
+	  else
+	    {
+	      // There is no previous queue item, seek to the start of the current item
+	      seek_ms = 0;
+	    }
+	}
+      else
+	{
+	  // We are more than 3 seconds into the playing track, seek to beginning of current track
+	  seek_ms = 0;
+	}
+    }
+  else if (seek_ms > 0 && seek_ms > pb_session.playing_now->len_ms)
+    {
+      // We are seeking beyond the current track, play the next track from the beginning
+      seek_queue_item = queue_item_next(pb_session.playing_now->item_id);
+      if (seek_queue_item)
```

Comment:
If there is no next item, then it will seek to a position that is beyond the length of playing_now - is that intentional?

Reply:
Yes, that was intentional. It will seek beyond the last track and playback will stop. I've added a log message and a comment for this case.